apps/worker/src/app/workflow/usecases/send-message/send-message.usecase.ts

```diff
@@ -306,51 +315,43 @@ export class SendMessage {
     });
     if (!subscriber) throw new PlatformException(`Subscriber not found with id ${job._subscriberId}`);
 
-    const isWorkflowPreferencesEnabled = await this.getFeatureFlag.execute(
-      GetFeatureFlagCommand.create({
-        userId: 'system',
-        environmentId: command.environmentId,
-        organizationId: command.organizationId,
-        key: FeatureFlagsKeysEnum.IS_WORKFLOW_PREFERENCES_ENABLED,
-      })
-    );
-
     /*
      * TODO: Remove this after we deprecate V1 preferences, global subscriber
```

Comment:
@rifont Can you please share more context on this? Do you see any issues with backwards compatibility between V1 and V2 after the removal of the feature flag?

Reply:
We haven't performed a migration of the `SubscriberPreferenceRepository` into the `Preferences` repository (see the current lookup in [get-subscriber-global-preference.usecase.ts here](https://github.com/novuhq/novu/blob/next/libs/application-generic/src/usecases/get-subscriber-global-preference/get-subscriber-global-preference.usecase.ts#L36)), therefore the current effect of this change in the absence of a migration will be that Subscribers who have modified Global Preferences will need to revisit the `<Inbox />` and set their preferences again for them to become active.

Today there are 118,719 global preferences set in Prod US, all of those Subscriber preferences would not be respected without a data migration.

Reply:
Regarding Workflow level preferences, this change is backwards compatible because [get-subscriber-template-preference.usecase.ts](https://github.com/novuhq/novu/blob/next/libs/application-generic/src/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts#L91) falls back to V1 preferences if V2 doesn't exist.

Reply:
I have applied the strangler pattern approach as listed above:
> keep the current Global Preferences lookup in place in the Send-Message usecase, then modify the [get-subscriber-global-preference.usecase.ts](https://github.com/novuhq/novu/blob/next/libs/application-generic/src/usecases/get-subscriber-global-preference/get-subscriber-global-preference.usecase.ts#L36) to fetch Subscriber Global preferences with a similar fallback approach as used for Subscriber Workflow preferences.

I will create a ticket for us to remove the V1 Preferences lookups in both of `GetSubscriberGlobalPreference` and `GetSubscriberTemplatePreference`, and remove this eager Global Preference lookup in `SendMessage` in 6 months time. That should give plenty of time for new preferences to be set so we can safely shift to V2 only Preference lookups.